radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyLedgerEntryStore.java

```diff
@@ -416,60 +394,52 @@ private void storeMetadata(Transaction dbTxn, LedgerAndBFTProof ledgerAndBFTProo
     final var nextEpoch = ledgerAndBFTProof.getProof().getEpoch() + 1;
 
     ledgerAndBFTProof
-        .getValidatorsSystemMetadata()
-        .ifPresent(
-            validatorSystemMetadata ->
-                storeEpochValidatorsSystemMetadata(dbTxn, nextEpoch, validatorSystemMetadata));
+        .getForksVotingResults()
+        .ifPresent(forksVotingResults -> storeForksVotingResults(dbTxn, forksVotingResults));
 
     ledgerAndBFTProof
-        .getNextForkHash()
-        .ifPresent(nextForkHash -> this.storeEpochForkHash(dbTxn, nextEpoch, nextForkHash));
+        .getNextForkName()
+        .ifPresent(nextForkName -> this.storeForkAtEpoch(dbTxn, nextEpoch, nextForkName));
   }
 
-  private void storeEpochValidatorsSystemMetadata(
-      Transaction dbTxn, long epoch, ImmutableList<RawSubstateBytes> validatorSystemMetadata) {
-    removeOldEpochValidatorsSystemMetadataEntries(dbTxn, epoch);
-    final var key = new DatabaseEntry(Longs.toByteArray(epoch));
-    validatorSystemMetadata.forEach(
-        data ->
-            validatorsSystemMetadataDatabase.put(dbTxn, key, new DatabaseEntry(data.getData())));
+  private void storeForksVotingResults(
+      Transaction dbTxn, ImmutableSet<ForkVotingResult> forksVotingResults) {
+    forksVotingResults.forEach(
```

Comment:
In cases like that, a simple for-each loop is more readable imho.

Reply:
I think it's a matter of taste, but in general we try to follow the functional style where possible

Reply:
I would suggest to be more flexible and use functional where it is more appropriate and not use it when it is not. I've seen @talekhinezh making similar suggestions before.

Reply:
why do you think it's not appropriate here?

Reply:
My comment was more related to the approach of trying to follow the functional style where possible. But in this case a simple for-each would be better imo. Streams seem more suitable for data processing pipelines.